### DISCUSSION TITLE
How to add custom autounattend.xml / additional files / folders

### DISCUSSION THREAD

**Discussion by igpit** [AveYo/MediaCreationTool.bat#134] [General]
Hello,

thx for a great script so far. We used to create ISO media and place a custom autounattend.xml and additional folders in the root directory, for extra update files, post-config, ... How could that be done with the existing script?

I know you pick up $OEM$ but where would/could the pre-configured autounattend.xml go inside that to be picked up during windows setup? 

I know you are reluctant to pick up additional folders, but like $OEM$ you could have/define a folder / entry point like "ADD_TO_ISO_ROOT" , if you see that folder, add everything inside to the root ISO/USB. 

Or is there a better way to go about adding things to the medium? 

Regards ;)

**Comment by AveYo** [comment]
I might add an option for custom folder.

Note that for 11 there is already an autounattend.xml added to bypass Home online account, but it's not added to the root directory because that would break normal setup within windows. Instead, it's added to boot.wim. Guess I could extend it to work on 10 as well, and add an option to pickup an already made one next to the script.

**Comment by igpit** [comment]
sounds great! you still planning to do it? just wondering cause issue got closed and no new code release... regards ;)

**Comment by AveYo** [comment]
Yeah, I'm still planning on. Will reopen the issue and ask for further input if needed when I get to it. Plate full atm.

**Comment by igpit** [comment]
thanks for reopening, looking forward to this ;)

currently we have a custom autounattend.xml and in there use 

`<RunSynchronousCommand> and <SynchronousCommand> `

referencing other files supplied on the install medium doing stuff.

**Comment by AveYo** [comment]
Yes, hopefully later today will update the script.
Will pick up a `$ISO$` folder next to the script, and add it's content to root of created media.
So the $OEM$ folder should now be placed in `$ISO$\sources\$OEM$` instead.
Script generated files (`AutoUnattend.xml`, `auto.cmd`, `EI.cfg`, `PID.txt`) will **not** override any that you have placed in `$ISO$` folder.
What do you think?

**Comment by igpit** [comment]
sounds great!

**Comment by AveYo** [comment]
I have finally implemented your request, as described above. Thank you!

**Comment by igpit** [comment]
tested and seems to work great. ty!
